src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomSetPropertyFunction.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+
+namespace Microsoft.PowerFx
+{
+    // Helper for SetPropertyFunction 
+    // Binds as: 
+    //    SetProperty(control.property:, arg:any)
+    // Invokes as:
+    //    SetProperty(control, "property", arg)
+    internal sealed class CustomSetPropertyFunction : TexlFunction
+    {
+        public override bool IsAsync => true;
+
+        public override bool IsSelfContained => false; // marks as behavior 
+
+        public override bool SupportsParamCoercion => false;
+
+        public Func<FormulaValue[], Task<FormulaValue>> _impl;
+
+        private readonly IEnumerable<CustomFunctionSignatureHelper> _argumentSignatures;
```

Comment:
> private readonly IEnumerable<CustomFunctionSignatureHelper> _argumentSignatures; [](http://example.com/codeflow?start=7&length=81)

This holds custom signatures,
each entry would map to a single overload.

If function doesn't have any overload, this will have single entry.